tests/characters/test_alice_can_grant_and_revoke.py

```diff
@@ -96,6 +102,37 @@ def test_federated_grant(federated_alice, federated_bob):
         assert kfrag == retrieved_kfrag
 
 
+@pytest.mark.usefixtures('federated_ursulas')
+def test_revocation(federated_alice, federated_bob):
+    m, n = 2, 3
+    policy_end_datetime = maya.now() + datetime.timedelta(days=5)
+    label = b"revocation test"
+
+    policy = federated_alice.grant(federated_bob, label, m=m, n=n, expiration=policy_end_datetime)
+
+    # Test that all arrangements are included in the RevocationKit
+    for node_id, arrangement_id in policy.treasure_map:
+        assert policy.revocation_kit[node_id].arrangement_id == arrangement_id
+
+    # Test revocation kit's signatures
+    for revocation in policy.revocation_kit:
+        assert revocation.verify_signature(federated_alice.stamp.as_umbral_pubkey())
+
+    # Test Revocation deserialization
+    revocation = policy.revocation_kit[node_id]
+    revocation_bytes = bytes(revocation)
+    deserialized_revocation = Revocation.from_bytes(revocation_bytes)
+    assert deserialized_revocation == revocation
+
+    # Attempt to revoke the new policy
+    failed_revocations = federated_alice.revoke(policy)
+    assert len(failed_revocations) == 0
```

Comment:
It would be helpful to see a test showing failures occurring, IMO.

Reply:
Read right below that. :)

Reply:
Yes - I mean I situation where some revocations fail

Reply:
... They did fail in that test, though. :confused:

Reply:
I guess I am misunderstanding something.  This line reads `assert len(failed_revocations) == 0` and you're telling me they failed?

Reply:
Read the statements right below that:

```python
# Try to revoke the already revoked policy
already_revoked = federated_alice.revoke(policy)
assert len(already_revoked) == 3
```

Reply:
Yes - but that tests something different.

Reply:
No, it tests the exact case you're asking for -- a failed revocation. If there is no KFrag to revoke, it's a failed revocation. You can see what the specific fail case was by checking the status code, whether it be a 404 or non-200.

Reply:
but this line reads `assert len(failed_revocations) == 0` .. it literally says there are 0 failed revocations. I dont understand how this constitutes the narrative I am describing.

Reply:
Then `already_revoked` could be changed to `failed_revocations` and it would satisfy you?

Reply:
Well - I was suggesting an alternate narrative - the policy has already been revoked in the existing test.  Another possibly common pathway is that the revocations fail the first time around, no?